mlflow/utils/autologging_utils/versioning.py

```diff
@@ -62,7 +63,16 @@ def is_flavor_supported_for_associated_package_versions(flavor_name):
         associated packages.
     """
     module_name = FLAVOR_TO_MODULE_NAME[flavor_name]
-    actual_version = importlib.import_module(module_name).__version__
+
+    try:
+        actual_version = importlib.import_module(module_name).__version__
+    except AttributeError:
+        try:
+            actual_version = importlib.metadata.version(module_name)
```

Comment:
Is `module_name` a package name? scikit-learn is a good example. The package name is `scikit-learn` while the module name is `sklearn`.

Reply:
That's fair. We can get package name using pypi index like what we do for requirement inference ([code](https://github.com/mlflow/mlflow/blob/master/mlflow/utils/requirements_utils.py#L511-L512)), but it doesn't sound worthwhile for this purpose. DSPy is the first case we saw `.__version__` doesn't work and we don't really need high accuracy in this function. I will remove this fallback for now.